### PR TITLE
Fix load exception on mono

### LINKED ISSUE #2: TypeLoadException w/ Mono 4.2.3
I'm trying to use Serilog.Exceptions 1.1.0 with Serilog 1.5.14 and Mono 4.2.3 but every time I try to run .Enrich.WithExceptionDetails using either the .NET 4.0 or 4.5 dlls (matching Serilog and Serilog.Exceptions of course) I'm getting a TypeLoadException. 

Here is a stack trace: 

```
   SetUp : System.TypeLoadException : Could not resolve type with token 01000058
  at Serilog.Exceptions.Destructurers.ExceptionEnricher..ctor (IEnumerable`1 destructurers) <0x41a1fe80 + 0x000e4> in <filename unknown>:0 
  at Serilog.Exceptions.Destructurers.ExceptionEnricher..ctor (Serilog.Exceptions.Destructurers.IExceptionDestructurer[] destructurers) <0x41a1fe50 + 0x00013> in <filename unknown>:0 
  at Serilog.Exceptions.Destructurers.ExceptionEnricher..ctor () <0x41a1fe10 + 0x0001f> in <filename unknown>:0 
  at Serilog.Exceptions.LoggerEnrichmentConfigurationExtensions.WithExceptionDetails (Serilog.Configuration.LoggerEnrichmentConfiguration loggerEnrichmentConfiguration) <0x41a1fb60 + 0x0005b> in <filename unknown>:0 
  at MyApp.InitializeGlobalLogger () <0x41a1a5d0 + 0x0014b> in <filename unknown>:0 
  at SetUpFixture..ctor () <0x41a1a580 + 0x0000b> in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoCMethod:InternalInvoke (System.Reflection.MonoCMethod,object,object[],System.Exception&)
  at System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) <0x4183a250 + 0x00045> in <filename unknown>:0 
```

and some version info:
```
Mono JIT compiler version 4.2.3 (Stable 4.2.3.4/832de4b Wed Mar 16 13:19:08 UTC 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          supported, not enabled.
	GC:            sgen
```

If Serilog.Exceptions uses P/Invokes then that is certainly the issue as these are not supported by Mono. Thanks for taking a look, and let me know if I can provide more information.

### PR DESCRIPTION
**Pull request by laedit**
Fix #2.
The dangerous types are loaded from string in order to avoid the load type exception.

I have added a test but it need a TravisCI build to check if it's work with mono.

### CODE REVIEW

File: Source/Serilog.Exceptions.Test/ExceptionDestructurerTest.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                Assert.Contains(targetTypes, t => t.FullName == "System.Diagnostics.Eventing.Reader.EventLogProviderDisabledException");
+                Assert.Contains(targetTypes, t => t.FullName == "System.Diagnostics.Eventing.Reader.EventLogReadingException");
+                Assert.Contains(targetTypes, t => t.FullName == "System.Management.Instrumentation.InstanceNotFoundException");
+                Assert.Contains(targetTypes, t => t.FullName == "System.Management.Instrumentation.InstrumentationBaseException");
+                Assert.Contains(targetTypes, t => t.FullName == "System.Management.Instrumentation.InstrumentationException");
+            }
+            else
+            {
+                // mono
+                Assert.Equal(118, targetTypes.Length);
```
Comment (RehanSaeed) [inline Source/Serilog.Exceptions.Test/ExceptionDestructurerTest.cs]: We might add other types. It's going to get annoying updating this number.

File: Source/Serilog.Exceptions/Destructurers/ExceptionDestructurer.cs [hunk 2]
```diff
@@ -184,9 +172,35 @@ public virtual Type[] TargetTypes
...
+                    {
+                        targetTypes.Add(type);
+                    }
+                }
+#endif
+                return targetTypes.ToArray();
             }
         }
 
+        private string[] GetDangerousTypes()
```
Comment (RehanSaeed) [inline Source/Serilog.Exceptions/Destructurers/ExceptionDestructurer.cs]: Can we add a comment about Mono not supporting these types please and/or maybe name the method to reflect this.

**Comment by RehanSaeed** [comment]
Looking very good. Only two minor comments and I'll get this merged and we can do another release.

I added an AppVeyor build earlier, I'm still working on perfecting it. Will probably add MyGet, so you can test the package before we push it to NuGet.

I saw your comment about Travis. It's probably worth adding that. I'm not certain how easy it is to use Travis with .NET Core projects though. Can I re-use my Build.ps1 PowerShell file?

File: Source/Serilog.Exceptions/Destructurers/ExceptionDestructurer.cs (hunk 2 above)
Comment (laedit) [inline Source/Serilog.Exceptions/Destructurers/ExceptionDestructurer.cs]: Thanks, I was struggling to find a good name bu wasn't thinking about Mono at all :smile:

**Comment by laedit** [comment]
I'll address these comments ASAP.

About building on TravisCI, sadly Powershell on Linux in not stable enough to allow that.
But apparently it can be done by adding a simple [`build.sh`](http://andrewlock.net/adding-travis-ci-to-a-net-core-app/) script alongside the mandatory `.travis.yml` configuration.

File: Source/Serilog.Exceptions.Test/ExceptionDestructurerTest.cs (hunk 1 above)
Comment (laedit) [inline Source/Serilog.Exceptions.Test/ExceptionDestructurerTest.cs]: I'll remove the length assert and just check if types are present or not.

**Comment by RehanSaeed** [comment]
I've added a MyGet feed, ready when you are.

**Comment by laedit** [comment]
I have fixed the comments, tell me if that's ok (or not).

I have also tested the package from AppVeyor's artifacts and [all is good](https://travis-ci.org/laedit/Serilog.Exceptions.Tests/builds/157156707)! 😄

**Comment by RehanSaeed** [comment]
Merged. I'll get it pushed as 2.0.1 tomorrow.

**Comment by RehanSaeed** [comment]
Thanks again. Am adding you to the contributors section on the front page.

**Comment by RehanSaeed** [comment]
NuGet updated.

**Pull request merged**